### PR TITLE
Create harvard-university-for-the-creative-arts(figures-and-illustrat…

### PR DESCRIPTION
**Pull request by ibadger**
…ions).csl

Hi
I have a valid CSL file for University for the Creative Arts which covers text sources in the Harvard style. As an creative arts university students are expected to submit a list of illustrations with each essay, these are in a numbered format. As a consequence I have created an additional style to cover images, I had seen that this was possible as there is a Chicago style which deals with figures and illustrations. I used that style as a starting point and have amended it to cover the Artwork and Maps styles as this will suit our needs. 

If I need to add anything or if there are any errors let me know

Ian

### CODE REVIEW

**Comment by rmzelle** [comment]
@ibadger, has your university published any referencing guidelines on this that we can link to?

And is the format different from that used by the Chicago style, or are you just extending it to other item types (like maps)?

**Comment by ibadger** [comment]
Our referencing guide is here: http://webdocs.ucreative.ac.uk/UCA%20Harvard%20referencing%20guide-1475663400023.pdf 

[UCA Harvard referencing guide-1475663400023.pdf](https://github.com/citation-style-language/styles/files/854323/UCA.Harvard.referencing.guide-1475663400023.pdf)

Sorry for not being very clear. I used the Chicago style as a base as it was the only style that I could find that dealt specifically with figures. It isn't really like the Chicago style, as it uses more fields and displays the fields in quite a different way.

I used the initial features of the style as it is similar to how we reference images as it displays the initial fields that we need, Figure number, creator of image, title of the image and then medium:

Figure 1. S Walker, _Dandelion flower bud_, Photograph; Transverse section through flower (infloresence) bud of dandelion, viewed with a light microscope, solarized..

This is similar to our style, which then has additions: figure number, creator of image, year, image title, medium, publication information e.g.:

Figure 5. Walker, S. (2003) _Dandelion flower bud_ [Photograph; Transverse section through flower (inflorescence) bud of dandelion, viewed with a light microscope, solarized.] At: http://images.wellcome.ac.uk/ (Accessed on 24 June 2009)

As it has a similar start is was easier for me to use to create the style that I needed, rather than create it from scratch.

Our style asks students to reference images used in their essay, they either reference images of illustrations from books, journals, online/adverts/stills/photographs or maps. As a consequence I have just used map and artwork as this covers everything.

If you need anything else just ask

Thanks for your help

Ian

**Comment by rmzelle** [comment]
@ibadger, I made a few minor edits (in part to align it more with the existing chicago-figures.csl style. Let me know if it's okay like this.

**Comment by rmzelle** [comment]
@ibadger, we don't necessarily inspect all commit messages, so it's better to leave your comments directly in the pull requests (like I'm doing right now).

> Hi
> 
> Thanks for making the changes and I'm happy with them.
> 
> I'm really sorry but suddenly realised that it didn't fit with how our referencing guide deals with the lack of a date and have since amended it so that it displays (s.d.) if there isn't a date in the record - hence them new commit.
> 
> Hope that this is ok, is there is a better way of adding this to the file just let me know.
> 
> Ian

File: university-for-the-creative-arts-figures.csl [hunk 1]
```diff
@@ -29,7 +29,16 @@
...
         <text variable="citation-number" suffix="."/>
         <text variable="title" font-style="italic"/>
-        <date date-parts="year" form="text" variable="issued"/>
+        <group>
+          <choose>
+            <if match="all" variable="issued">
+              <date date-parts="year" form="text" variable="issued"/>
+            </if>
+            <else>
+              <text value="s.d."/>
```
Comment (rmzelle) [inline university-for-the-creative-arts-figures.csl]: @ibadger, we greatly prefer the use of CSL terms wherever possible. You can replace `<text value="s.d."/>` by `<text term="no date" form="short"/>`. You would also have to redefine the translation of this term in the locale section at the top of the style with `<term name="no date" form="short">s.d.</term>`.

(see https://github.com/citation-style-language/locales/blob/b01a5e4d420a4c0e446625c271144c52b0ab0a76/locales-en-GB.xml#L45 for the default translation of this term)

File: university-for-the-creative-arts-figures.csl [hunk 2]
```diff
@@ -29,7 +29,16 @@
         <text term="figure" form="symbol" text-case="capitalize-first"/>
         <text variable="citation-number" suffix="."/>
         <text variable="title" font-style="italic"/>
-        <date date-parts="year" form="text" variable="issued"/>
+        <group>
```
Comment (rmzelle) [inline university-for-the-creative-arts-figures.csl]: @ibadger, I don't think you need the extra wrapping `<group/>` here.

**Comment by rmzelle** [comment]
> Hope that this is ok, is there is a better way of adding this to the file just let me know.

No, what you did is exactly the right way. As long as we haven't accepted the pull request, you can keep making changes.

**Comment by rmzelle** [comment]
(@adam3smith, happy with this?)

**Comment by ibadger** [comment]
Hi

Sorry for all the mess on here. I've made the edits, in the process I messed a couple of other things up. I tend to work with the csl visual editor and ended up editing an older version, and then having other issues.

Let me know if anything else should be amended

Ian

**Comment by rmzelle** [comment]
Great, thanks!

**Pull request merged**